### PR TITLE
Small molecule chromatogram library column renaming

### PR DESCRIPTION
**Pull request by labkey-jeckels**
#### Rationale
We can freely improve naming in the small molecule chromatogram schema for CLIB files since they're not in use in the wild yet. Also, we can consolidate the DriftTimeXXX columns on PrecursorChromInfo and TransitionChromInfo into the newer IonMobilityXXX columns

#### Changes
* Rename from CustomIonName->MoleculeName and IonFormula->ChemicalFormula
* Extract Adduct part of MoleculePrecursor formula
* Migrate any existing Panorama DriftTimeXXX values into IonMobilityXXX columns and drop the old ones

### CODE REVIEW

File: src/org/labkey/targetedms/parser/SkylineDocumentParser.java [hunk 1]
```diff
@@ -2195,13 +2195,26 @@ private PrecursorChromInfo readPrecursorChromInfo(XMLStreamReader reader) throws
...
+
+        // Support old Skyline documents that used "drift time" instead of "ion mobility"
+        Double driftTimeMs1 = XmlUtil.readDoubleAttribute(reader, "drift_time_ms1");
+        Double driftTimeWindow = XmlUtil.readDoubleAttribute(reader, "drift_time_window");
+        Double driftTimeFragment = XmlUtil.readDoubleAttribute(reader, "drift_time_fragment");
+
+        if (driftTimeMs1 != null || driftTimeWindow != null || driftTimeFragment != null)
+        {
+            chromInfo.setIonMobilityMs1(driftTimeMs1);
+            chromInfo.setIonMobilityMs1(driftTimeFragment);
```
Comment (labkey-ankurj) [inline src/org/labkey/targetedms/parser/SkylineDocumentParser.java]:
```suggestion
            chromInfo.setIonMobilityFragment(driftTimeFragment);
```

**Review by labkey-ankurj — CHANGES_REQUESTED** [review]

File: src/org/labkey/targetedms/chromlib/LibMolecule.java [hunk 2]
```diff
@@ -3,30 +3,30 @@
 public class LibMolecule extends AbstractLibMolecule<LibMoleculePrecursor>
 {
     private Integer _moleculeListId;
-    private String _ionFormula;
-    private String _customIonName;
+    private String _chemicalFormula;
+    private String _moleculeName;
```
Comment (labkey-ankurj) [inline src/org/labkey/targetedms/chromlib/LibMolecule.java:7]: does this need to be renamed as fragmentName?

File: src/org/labkey/targetedms/chromlib/Constants.java [hunk 3]
```diff
@@ -121,8 +121,8 @@ private Constants() {}
         CalcNeutralMass("DOUBLE NOT NULL"),
         NumMissedCleavages("INTEGER NOT NULL"),
 
-        IonFormula("VARCHAR(100)"),
-        CustomIonName("VARCHAR(100)"),
+        ChemicalFormula("VARCHAR(100)"),
+        MoleculeName("VARCHAR(100)"),
```
Comment (labkey-ankurj) [inline src/org/labkey/targetedms/chromlib/Constants.java:125]: I suggest to remove this and use FragmentName defined at L181 in MoleculeTransitionColumn. FragmentName column

File: src/org/labkey/targetedms/chromlib/Constants.java [hunk 4]
```diff
@@ -710,8 +707,8 @@ public String definition()
     {
         Id(Column.Id),
         MoleculeListId(Column.MoleculeListId),
-        IonFormula(Column.IonFormula),
-        CustomIonName(Column.CustomIonName),
+        IonFormula(Column.ChemicalFormula),
+        CustomIonName(Column.MoleculeName),
```
Comment (labkey-ankurj) [inline src/org/labkey/targetedms/chromlib/Constants.java]: may be use Column.FragmentName

**Review by labkey-ankurj — APPROVED** [review]
couple of additional renaming comments

File: src/org/labkey/targetedms/chromlib/LibMolecule.java (hunk 2 above)
Comment (labkey-jeckels) [reply src/org/labkey/targetedms/chromlib/LibMolecule.java:7]: Brian requested that we keep the molecule one as MoleculeName and call the MoleculeTransition one as FragmentName, which I think is the case as currently coded.

**Comment by labkey-ankurj** [comment]
ah yes, got it. thanks

File: src/org/labkey/targetedms/chromlib/Constants.java (hunk 4 above)
Comment (labkey-jeckels) [reply src/org/labkey/targetedms/chromlib/Constants.java]: This pattern's a little clunky, but the two names should agree with each other. I found a couple of cases that were out of sync, including this one.

**Pull request merged**